server/eclipse-project/src/main/java/us/freeandfair/corla/query/ExportQueries.java

```diff
@@ -269,4 +284,63 @@ public static long custCopyOut(final String sql, OutputStream to, CopyManager cm
     }
   }
 
+  /**
+   * This function deals, somewhat inelegantly, with the problem that the ContestResult data structure
+   * used in most queries does not have correct values for things like winners, losers, margin, and
+   * diluted margin. This sets them manually from the GenerateAssertionsSummary table, then flushes
+   * the database so that the csv reports, which are based on database queries, get the right
+   * values from the contest_result table.
+   * @param s The current Hibernate session.
+   */
+  private static void updateIRVContestResults(final Session s) {
+    final String prefix = "[updateIRVContestResults]";
+    LOGGER.debug(String.format("%s %s.", prefix,
+        "Updating IRV contest results from generate assertions summary"));
+
+    final List<ComparisonAudit> comparisonAudits = ComparisonAuditQueries.sortedList();
+
+    for(final ComparisonAudit ca : comparisonAudits) {
+      if(ca instanceof IRVComparisonAudit) {
+        final Set<String> choices = new HashSet<>();
+        // Get the choices for the contest. These should be the same for all the contests, but
+        // gather the whole set from all of them just in case.
```

Comment:
Would it make sense to iterate over final Contest contest? It looks like line 309 is updating the choices set defined on line 304, yet choices has been defined final. (I think the rules are a bit weird/counter intuitive when objects are made final).

Reply:
Yes it's updating the object, but it's not reassigning the value, so 'final' is OK.